variables.tf

```diff
@@ -248,3 +248,107 @@ variable "at_cos_bucket_endpoint" {
   description = "An endpoint for the COS bucket for the Activity Tracker archive. Pass either the public or private endpoint (Only required when var.enable_archive and var.activity_tracker_provision are true)"
   default     = null
 }
+
+########################################################################
+# Activity Tracker Event Routing
+#########################################################################
+
+# COS Targets
+variable "cos_targets" {
```

Comment:
I think you should rename this to be non-plural, or make the type an actual array so that multiple can be supplied.
By saying "targets" on each one of these, that usually implies multiple targets, but I don't see an array in the type, implying there is only one target?
There is also a for_each used in the main.tf code that is not really used as a for loop, as this data type is not an array.